File: shop-test/pages/CheckoutPage.ts
import type { Page } from "@playwright/test";

export class CheckoutPage{
    constructor(private readonly page: Page){}
    private placeOrderButton= ()=> this.page.getByRole("button", {name: "Place order"});
    private viewOrderButton= ()=> this.page.getByRole("button", {name: "View orders"});

    async open(){
        await this.page.goto('/checkout');
    }

    async placeOrder(){
        // const apiCheckout= await this.page.waitForResponse(
        //     r=> r.url().includes('/api/orders') &&
        //     r.status() ===200
        // ) 
        await this.placeOrderButton().click()
        // await apiCheckout;
    }

    async vieweOrder(){
        const apiCheckout= this.page.waitForResponse(
            r=> r.url().includes('/api/orders') &&
            r.status() ===200
        ) 
        await this.viewOrderButton().click()
        await apiCheckout;
    }

    results(){
        return this.page.getByTestId("cart-count");
    }

    async orderID(){
        const [text]=await this.page.locator(".confirmation-panel > p >strong").allTextContents()
        return text;
    }

}

File: Selenide_Project/src/test/java/com/ust/sdet/pages/CheckoutPage.java
package com.ust.sdet.pages;

import com.codeborne.selenide.SelenideElement;

import static com.codeborne.selenide.Selenide.*;

public class CheckoutPage {

    private SelenideElement checkoutTitle = $("#checkout-title");
    private SelenideElement placeOrderButton = $("[data-test='place-order']");
    private SelenideElement viewOrdersButton = $(".button.primary");

    public SelenideElement checkoutTitle() {
        return checkoutTitle;
    }

    public CheckoutPage placeOrder() {
        placeOrderButton.click();
        return this;
    }

    public OrdersPage viewOrders() {
        viewOrdersButton.click();
        return page(OrdersPage.class);
    }
}